sdk/metric/doc.go

```diff
@@ -39,6 +39,30 @@
 // Meter.RegisterCallback and Registration.Unregister to add and remove
 // callbacks without leaking memory.
 //
+// # Cardinality Limits
+//
+// Cardinality refers to the number of unique attribute combinations for a single
+// metric within one collection (export) interval. High cardinality can lead to
+// excessive memory usage, increased storage costs, and backend performance issues.
+//
+// The OpenTelemetry Go Metric SDK does not enforce a cardinality limit by default.
+// A limit can be configured using the [WithCardinalityLimit]
+// option.
```

Comment:
```suggestion
// Currently, the OpenTelemetry Go Metric SDK does not enforce a cardinality limit by default
// (note that this may change in a future release). Use [WithCardinalityLimit] to set the
// cardinality limit as desired.
```